src/js/core/range.js

```diff
@@ -468,7 +463,7 @@ define([
             topAncestor = ancestors[ancestors.length - 2] || sc.childNodes[so];
           }
         } else {
-          topAncestor = sc.childNodes[so - 1];
+          topAncestor = sc.childNodes[so > 0 ? so - 1 : 0];
```

Comment:
@easylogic Please review this code.
code for `body inline case 1` ~ `body inline case 4`
https://github.com/summernote/summernote/pull/846/files#diff-458c62a88878b655f75be1363f0c4652R162